pkg/util/objectwatcher/objectwatcher.go

```diff
@@ -121,6 +121,10 @@ func (o *objectWatcherImpl) retainClusterFields(desired, observed *unstructured.
 	// controllers in a member cluster.  It is still possible to set the fields
 	// via overrides.
 	desired.SetFinalizers(observed.GetFinalizers())
+
+	// Retain ownerReferences since they will typically be set by controllers in a member cluster.
+	desired.SetOwnerReferences(observed.GetOwnerReferences())
```

Comment:
That makes sense. Here we don't `merge` the owner reference because we usually won't set owner reference in the resource template.

@mrlihanbo Could you please double confirm it?